core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -146,4 +152,15 @@ public SparkSqlDialect(SqlDialect.Context context) {
       super.unparseCall(writer, call, leftPrec, rightPrec);
     }
   }
+
+  @Override public @Nullable SqlNode getCastSpec(RelDataType type) {
+    if (type instanceof BasicSqlType) {
+      if (type.getSqlTypeName() == SqlTypeName.VARCHAR) {
```

Comment:
minor: can use a `switch case` block or use a single `if` judgment instead of `nested ifs`?

Reply:
Thanks for the review @YiwenWu . I wil use if to replace nested ifs.

Reply:
Hi @YiwenWu , I have fix this.